addons/l10n_de_pos_cert/models/pos_config.py

```diff
@@ -0,0 +1,166 @@
+# -*- coding: utf-8 -*-
+
+from odoo import models, fields, _, api
+from odoo.exceptions import ValidationError, UserError
+import requests
+from requests.exceptions import ConnectTimeout
+import uuid
+
+
+class PosConfig(models.Model):
+    _inherit = 'pos.config'
+    fiskaly_tss_id = fields.Char(string="TSS ID", readonly=True)
+    fiskaly_client_id = fields.Char(string="Client ID", readonly=True)
+    create_tss_flag = fields.Boolean(default=False)
+    is_company_country_germany = fields.Boolean(compute='_compute_is_company_country_germany')
+
+    @api.depends('company_id.country_id')
+    def _compute_is_company_country_germany(self):
+        if self.company_id.country_id and self.company_id.country_id.code == "DE":
+            self.is_company_country_germany = True
+        else:
+            self.is_company_country_germany = False
+
+    def _check_fiskaly_key_secret(self):
+        if not self.company_id.fiskaly_key or not self.company_id.fiskaly_secret:
+            raise UserError(_("You have to set your Fiskaly key and secret in your company settings."))
+
+    def _check_fiskaly_tss_client_ids(self):
+        if not self.fiskaly_tss_id or not self.fiskaly_client_id:
+            raise UserError(_("You have to set your Fiskaly TSS ID and Client ID in your PoS settings."))
+
+    def open_ui(self):
+        if not self.company_id.country_id:
+            raise UserError(_("You have to set a country in your company setting."))
+        if self.company_id.country_id.code == "DE":
+            self._check_fiskaly_key_secret()
+            self._check_fiskaly_tss_client_ids()
+        return super(PosConfig, self).open_ui()
+
+    @api.model
+    def create(self, values):
+        res = super(PosConfig, self).create(values)
+        if 'create_tss_flag' in values and values['create_tss_flag']:
+            res.create_tss_process()
+        return res
+
+    def write(self, values):
+        res = super(PosConfig, self).write(values)
+        if 'create_tss_flag' in values and values['create_tss_flag']:
+            self.create_tss_process()
+        return res
+
+    def unlink(self):
+        if self.create_tss_flag:
+            cache = {
+                'fiskaly_tss_id': self.fiskaly_tss_id,
+                'fiskaly_secret': self.company_id.fiskaly_secret,
+                'fiskaly_key': self.company_id.fiskaly_key
+            }
+            res = super(PosConfig, self).unlink()
+            self.delete_tss(cache)
+            return res
+        else:
+            return super(PosConfig, self).unlink()
+
+    def create_tss_process(self):
```

Comment:
make private (and make multi (loop for config in self))
`def _create_tss_process(self)`